### ISSUE TITLE
Translate between domains with functors

### ISSUE THREAD

**Issue by hypotext** [penrose/penrose#251]
Let's say we are working with a domain that involves, say, loops on surfaces. We know that there is a correspondence between loops on surfaces and groups, so we might want to use the group style.

If we could write down the mapping/correspondence between the two domains, then we could use the group style out of the box. This mapping could be implemented with a string find-and-replace.

For example (for a different example):
```
Point <-> Table
Line <-> Mug
...
```

Basically, we are implementing functors (from category theory) in Penrose.

(This came up in discussion with @keenancrane)
